### PR TITLE
Remove unused GETFIELD/SETFIELD helpers

### PR DESCRIPTION
**Pull request by EgorBo**
Also, JIT used to pretend that `CORINFO_HELP_UNBOX_NULLABLE` returned a TYP_STRUCT and had some special handling around multi-reg return. While in fact the helper is VOID

### CODE REVIEW

File: src/coreclr/inc/readytorun.h [hunk 1]
```diff
@@ -20,7 +20,7 @@
 // If you update this, ensure you run `git grep MINIMUM_READYTORUN_MAJOR_VERSION`
 // and handle pending work.
 #define READYTORUN_MAJOR_VERSION 10
-#define READYTORUN_MINOR_VERSION 0x0001
+#define READYTORUN_MINOR_VERSION 0x0002
```
Comment (jkotas) [inline src/coreclr/inc/readytorun.h:23]: This should not be necessary. The helpers that you are deleting were never exposed as R2R helpers.

Comment (EgorBo) [reply src/coreclr/inc/readytorun.h]: Thanks, reverted

**Comment by EgorBo** [comment]
@MihuBot

File: src/coreclr/jit/importer.cpp [hunk 2]
```diff
@@ -10100,22 +10100,38 @@ void Compiler::impImportBlockCode(BasicBlock* block)
...
+                        //
+                        unsigned resultTmp = lvaGrabTemp(true DEBUGARG("Nullable<T> tmp"));
+                        lvaSetStruct(resultTmp, resolvedToken.hClass, false);
+
+                        GenTreeLclFld* resultAddr = gtNewLclAddrNode(resultTmp, 0);
+                        // NOTE: it's fine for op2 to be evaluated before op1
+                        GenTreeCall* helperCall = gtNewHelperCallNode(helper, TYP_VOID, resultAddr, op2, op1);
+                        impAppendTree(helperCall, CHECK_SPILL_ALL, impCurStmtDI);
+                        op1 = gtNewLclvNode(resultTmp, TYP_STRUCT);
                     }
```
Comment (jakobbotsch) [inline src/coreclr/jit/importer.cpp:10129]: Can you post diffs of this change alone?

**Review by jkotas — APPROVED** [review]

File: src/coreclr/jit/importer.cpp (hunk 2 above)
Comment (EgorBo) [reply src/coreclr/jit/importer.cpp:10129]: Sure will post here, I expect a few regressions, but this path is expected to be on a cold path with https://github.com/dotnet/runtime/issues/104954

Comment (EgorBo) [reply src/coreclr/jit/importer.cpp:10129]: Jit-diff (linux-x64): https://github.com/MihuBot/runtime-utils/issues/535

**Comment by jakobbotsch** [comment]
> Also, JIT used to pretend that `CORINFO_HELP_UNBOX_NULLABLE` returned a TYP_STRUCT and had some special handling around multi-reg return. While in fact the helper is VOID

This is actually an optimization done by the importer. It avoids introducing unnecessary copies by passing the address of the store destination of the `unbox` operation directly as the address to the helper call, in the same way as return buffers from regular calls are handled.

I'm fine with simplifying all of this, especially if it's going to be on the cold path, just wanted to add some context and add it here for posterity.

**Comment by EgorBo** [comment]
> > Also, JIT used to pretend that `CORINFO_HELP_UNBOX_NULLABLE` returned a TYP_STRUCT and had some special handling around multi-reg return. While in fact the helper is VOID
> 
> This is actually an optimization done by the importer. It avoids introducing unnecessary copies by passing the address of the store destination of the `unbox` operation directly as the address to the helper call, in the same way as return buffers from regular calls are handled.
> 
> I'm fine with simplifying all of this, especially if it's going to be on the cold path, just wanted to add some context and add it here for posterity.

Yeah it's understandable, I presume it's better to use a separate node/GT_INTRINSIC for such case to avoid dealing with ABI stuff too early?

Here are the SPMI diffs on win-x64 (although, I couldn't download libraries.pmi.tests collection): https://gist.github.com/EgorBo/6d6f29c571b09f058ab3f2f80ba6e70a not too bad IMO

**Comment by EgorBo** [comment]
@jakobbotsch does the jit part look good otherwise? I've applied the `lvHasLdAddrOp` fix

**Review by jakobbotsch — APPROVED** [review]
LGTM, nice clean up!

**Pull request merged**